sov-modules/sov-state/src/storage.rs

```diff
@@ -1,25 +1,70 @@
 use std::sync::Arc;
 
+use sovereign_sdk::serial::Encode;
+
+use crate::{utils::AlignedVec, Prefix};
+
 // `Key` type for the `Storage`
 #[derive(Clone, PartialEq, Eq)]
 pub struct StorageKey {
-    pub key: Arc<Vec<u8>>,
+    key: Arc<Vec<u8>>,
+}
+
+impl StorageKey {
+    pub fn key(&self) -> Arc<Vec<u8>> {
+        self.key.clone()
+    }
+}
+
+impl AsRef<Vec<u8>> for StorageKey {
+    fn as_ref(&self) -> &Vec<u8> {
+        &self.key
+    }
+}
+
+impl StorageKey {
+    // Creates a new prefixed StorageKey.
+    pub fn new<K: Encode>(prefix: &Prefix, key: K) -> Self {
+        let mut encoded_key = Vec::default();
+        key.encode(&mut encoded_key);
+
+        let encoded_key = AlignedVec::new(encoded_key);
+
+        let full_key = Vec::<u8>::with_capacity(prefix.len() + encoded_key.len());
+        let mut full_key = AlignedVec::new(full_key);
+        full_key.extend(prefix.as_aligned_vec());
+        full_key.extend(&encoded_key);
```

Comment:
(non-blocking) It seems like we should be able to do this without encoding the key separately and then copying. Maybe we should add a `size_hint` method to the key type?

Reply:
Resolved offline, we are skipping it for now.